File: 백준/Silver/10773. 제로/제로.py
import sys
input = sys.stdin.readline

# 실제 실행되는 코드: sys.stdin.readline().strip()
K = int(input().strip())

# 정수를 저장할 스택
stack = list()

for _ in range(K):
  number = int(input().strip())
  #모든 수를 저장하는데 입력이 0이면 마지막 값을 삭제
  if number != 0:
    stack.append(number)
  elif number == 0:
    stack.pop()
  
# 모든 수의 합을 출력
# n = len(stack)
# total = 0
# for i in range(n):
#   total += stack[i]

print(sum(stack))

File: 백준/Silver/10773. 제로/제로.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(br.readLine());

        Stack<Integer> stack = new Stack<>();
        for (int i = 0; i < n; i++) {
            int num = Integer.parseInt(br.readLine());
            if (num == 0) {
                stack.pop();
            }
            else {
                stack.push(num);
            }
        }
        int sum = 0;
        for (int i : stack) {
            sum += i;
        }

        bw.write(String.valueOf(sum));
        bw.flush();
        bw.close();
        br.close();

    }
}